### PR TITLE
Migrate core/keyboard_nav/ast_node.js to goog.module syntax

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test` locally already.

### Resolves

Part of #5026

### Proposed Changes

Converts `core/keyboard_nav/ast_node.js` to `goog.module` syntax.

### CODE REVIEW

File: core/keyboard_nav/ast_node.js [hunk 1]
```diff
@@ -10,38 +10,38 @@
...
-goog.require('Blockly.utils.Coordinate');
-
-goog.requireType('Blockly.Block');
-goog.requireType('Blockly.Connection');
-goog.requireType('Blockly.Field');
-goog.requireType('Blockly.IASTNodeLocation');
-goog.requireType('Blockly.IASTNodeLocationWithBlock');
-goog.requireType('Blockly.Input');
-goog.requireType('Blockly.Workspace');
+const Block = goog.requireType('Blockly.Block');
```
Comment (maribethb) [inline core/keyboard_nav/ast_node.js:16]: I think for now you need to add the line disabling the lint warning (sorry, I'm working on it)

File: core/keyboard_nav/ast_node.js [hunk 2]
```diff
@@ -412,29 +415,30 @@ Blockly.ASTNode.prototype.findPrevForField_ = function() {
...
   if (!curLocation || !curLocation.workspace) {
     return null;
   }
-  var curRoot = curLocation.getRootBlock();
-  var topBlocks = curRoot.workspace.getTopBlocks(true);
-  for (var i = 0, topBlock; (topBlock = topBlocks[i]); i++) {
+  const curRoot = curLocation.getRootBlock();
+  const topBlocks = curRoot.workspace.getTopBlocks(true);
+  let i = 0, topBlock;
+  for (; (topBlock = topBlocks[i]); i++) {
```
Comment (maribethb) [inline core/keyboard_nav/ast_node.js:434]: Is this what it's supposed to be? I know this technically works, but this is really weird looking to me though and I thought `for (let i = 0, topBlock; topBlock = topBlocks[i]; i++)` was supposed to work? (there are several like this in this file)

File: core/keyboard_nav/ast_node.js [hunk 3]
```diff
@@ -444,125 +448,128 @@ Blockly.ASTNode.prototype.navigateBetweenStacks_ = function(forward) {
...
 
 /**
  * Find the element to the right of the current element in the AST.
- * @return {Blockly.ASTNode} An AST node that wraps the next field, connection,
+ * @return {ASTNode} An AST node that wraps the next field, connection,
  *     block, or workspace. Or null if there is no node to the right.
  */
-Blockly.ASTNode.prototype.next = function() {
+ASTNode.prototype.next = function() {
+  let connection, block, nextConnection, targetConnection;
```
Comment (maribethb) [inline core/keyboard_nav/ast_node.js:545]: I think declaring multiple variables in one line is disallowed: https://google.github.io/styleguide/jsguide.html#features-one-variable-per-declaration (which might conflict with the array advice..... in which case I would probably convert those arrays to more normal style of arrays `for (let i = 0, i < a.length; i++) {`)

File: core/keyboard_nav/ast_node.js (hunk 2 above)
Comment (moniika) [reply core/keyboard_nav/ast_node.js:434]: This is what my IDE automatically updated. I'll look into it.

File: core/keyboard_nav/ast_node.js (hunk 3 above)
Comment (moniika) [reply core/keyboard_nav/ast_node.js:545]: Noted. I try something else that follows the style guide. I was getting a different warning about declaring variables in a case block before, which is why i put them all up here.

File: core/keyboard_nav/ast_node.js (hunk 2 above)
Comment (cpcallen) [reply core/keyboard_nav/ast_node.js:434]: The IDE is probably worried that `i` or `topBlock` might be used after the end of the loop body—which is possible with a `var` (function scoped), but not with `let` (scoped only to the loop).

A bit disappointing it doesn't actually check, though.

File: core/keyboard_nav/ast_node.js (hunk 3 above)
Comment (cpcallen) [reply core/keyboard_nav/ast_node.js:545]: If it helps, you can (and [are allowed](https://google.github.io/styleguide/jsguide.html#formatting-switch-statements)) to put a block in each `case`:
```JS
switch (x) {
  case y: {
    // body;
    break;  // or return;
  }
  // etc.
}
```
…though admittedly it does look a little weird.

Comment (moniika) [reply core/keyboard_nav/ast_node.js:545]: I've added blocks to each `case` that require lexical scoping, but I'm not sure if I should add it to all cases if some of them need it.

For now I've only added it to those that need it, but I could add to the rest.

**Review by maribethb — APPROVED** [review]

**Pull request merged**